site/app/templates/calendar/Calendar.twig

```diff
@@ -78,11 +78,38 @@
         <div class="cal-legend">
             {% if show_all_cookie == 1 %}
                 {% for course in course_names %}
-                    <div name="{{course.course_name|replace({' ':''})}}" class="legend-color"></div>
-                    <p for="{{course.course_name|replace({' ':''})}}-color" class="legend-label">{{course.display_name}}</p>
+                <div 
+                    name="{{course.course_name|replace({' ':''})}}" 
+                    class="legend-color" 
+                    onclick="filter_course({{course.course_name}}, {{course.display_name}})" 
```

Comment:
You need to put quotes around these values so that they are valid JS strings.

```suggestion
                    onclick="filter_course('{{course.course_name}}', '{{course.display_name}}')" 
```